### PR TITLE
[CR] json/tools/iuse: cell phone to drain charges while on

### LINKED ISSUE #16634: Cell phones don't require batteries
Given that they tell time and can be alarm clocks i feel like im cheating to use it.

### PR DESCRIPTION
**Pull request by keyspace**
Closes #16634.

### Summary

Previously, `cell_phone` was an item that only drained batteries when activated (mutated into a `cell_phone_flashlight`).

* It was implemented completely in JSON.
* `cell_phone_flashlight` would provide a modest amount of light (`LIGHT_20`, 3x3),
* that would last a rather long time (~20 batteries per hour, 5 hours max).
* It had a minor bug of not having `WATCH` and `ALARMCLOCK` (which the "base" `cell_phone` did).
* The `ILLITERATE` character trait was not taken into account.

In this PR (which I'm not too proud of, see **bold** below):

* Battery draining and "flashlight app" switching is moved into an `iuse` method.
* Light level is not changed.
* When in "stand by" mode, the battery is drained slowly (1 charge per hour, ~4 days max). When the "flashlight app" is activated, the drain is significant (1 charge per minute, less than 2 hours max).
* The flags are fixed so that the phone only shows time when on. 
* `ILLITERATE` characters can't turn the "flashlight" on/off, although they can turn the phone itself on/off.

(Tangent: `ILLITERATE` doesn't prevent use of watches to tell time.)

### Implementation details

Implemented as an `iuse` (instead of pure JSON), since:

* multiple `transform`s for a single item (tried this approach first) result in only the first being available in-game;
* `ILLITERATE` trait has to be taken into account.

All three states (off/on/flashlight) are defined in JSON, since I couldn't figure out how to set `LIGHT_*` flag on an active item in a way that would actually produce light in-game. The "off" `cell_phone` uses a JSON `transform`, `cell_phone_{on,flashlight}` both use `iuse::cell_phone_on()`. **This mixing of data-driven and code-driven is not pretty.** :/

Other notes:

* `cell_phone` now has `rand_charges` instead of `initial_charges`;
* `flags` are specified explicitly for all JSON items;
* **EDIT: `LIGHT_20` appears in both JSON and `iuse` method;** :/
* **`cell_phone_flashlight` shows as "cell phone (on) (flashlight) (active)".** :/

-----

EDIT: bold

### CODE REVIEW

**Comment by BrettDong** [comment]
JSON linting suggests the following format of `data/json/items/tools.json`:
```JSON
    "id": "cell_phone",
    "type": "TOOL",
    "name": "cell phone (off)",
    "name_plural": "cell phones (off)",
    "description": "This is a typical smartphone that features an LED for use as a flash.  Using this will turn it on, assuming it is charged with batteries.",
    "weight": 226,
    "volume": 1,
    "price": 10000,
    "price_postapoc": 2000,
    "material": [ "plastic", "aluminum" ],
    "symbol": ";",
    "color": "light_gray",
    "ammo": "battery",
    "rand_charges": [ 20, 40, 60, 80, 100 ],
    "max_charges": 100,
    "use_action": {
      "target": "cell_phone_on",
      "msg": "The cell phone powers on.",
      "active": true,
      "need_charges": 1,
      "need_charges_msg": "The cell phone's batteries are dead.",
      "type": "transform"
    }
```

File: src/iuse.cpp [hunk 1]
```diff
@@ -7879,3 +7879,57 @@ int iuse::washclothes( player *p, item *it, bool, const tripoint& )
 
     return it->type->charges_to_use();
 }
+
+int iuse::cell_phone_on( player *p, item *it, bool t, const tripoint& )
+{
+    const std::string led_on = "LIGHT_20";
+    
+    if (t) {
```
Comment (BrettDong) [inline src/iuse.cpp]: Code style should be `if( t ) {`

**Comment by keyspace** [comment]
Thanks, will fix ASAP.

Still wondering if mixing data-driven `transform` and code-driven `convert` via `iuse` is acceptable.

**Comment by keyspace** [comment]
Taking into consideration the 3 issues with this (bold in preamble), IMO best leave the phone as-is for now.

Closing.

**Pull request closed without merging**